Core/Core/Discussions/DiscussionTopic.swift

```diff
@@ -96,6 +97,10 @@ public final class DiscussionTopic: NSManagedObject, WriteableModel {
         }
         model.canUnpublish = item.can_unpublish != false
         model.context = item.html_url.flatMap { Context(path: $0.path) }
+        if let contextCode = item.context_code {
+            let context = Context(canvasContextID: contextCode.value)
+            model.courseID = context?.id
+        }
```

Comment:
Does this end up being different from the `context` property above, which could already be `contextType == .course`?